### ISSUE TITLE
Question: Is there a way to modify this code to commit two or more files to a repo?

### ISSUE THREAD

**Issue by BarbourSmith** [gr2m/octokit-create-pull-request#6]
I've been struggling with trying to commit multiple files to a repo at the same. My files could be large enough to require being in a blob.

I initially created [this StackOverflow question and answer](https://stackoverflow.com/questions/55548359/how-to-commit-files-larger-than-1mb-to-github-using-the-github-api-and-octokit-j) to answer the question of commiting just one file to a repo, but now I am having issues with that code where git.getRef will return the correct reference the first time it is called, but then the reference never updates even when I can see it updating if I manually check https://api.github.com/repos/BarbourSmith/test2/git/refs/heads/master . I'm pretty sure the issue is on my end, not a bug but I can't find what is wrong so I am looking for an example of how to do things right.

TLDR: Is there a way for me to modify this library to take n files and commit them to master instead of creating a pull request?

Thanks for your time in reading!

**Comment by BarbourSmith** [comment]
It was pretty simple to figure out, thank you so much for making this library which served as a template.

If you ever feel like it, I bet there are a lot of other folks out there who would love a plugin for commiting several files to github through octokit. I never would have figured it out without this example.

For anyone who stumbles across this the code which worked for me was:

```
async createCommit (octokit, { owner, repo, base, changes }) {
      let response;

      if (!base) {
        response = await octokit.repos.get({ owner, repo })
        base = response.data.default_branch
      }

      response = await octokit.repos.listCommits({
        owner,
        repo,
        sha: base,
        per_page: 1
      })
      let latestCommitSha = response.data[0].sha
      const treeSha = response.data[0].commit.tree.sha
      
      response = await octokit.git.createTree({
        owner,
        repo,
        base_tree: treeSha,
        tree: Object.keys(changes.files).map(path => {
          return {
            path,
            mode: '100644',
            content: changes.files[path]
          }
        })
      })
      const newTreeSha = response.data.sha

      response = await octokit.git.createCommit({
        owner,
        repo,
        message: changes.commit,
        tree: newTreeSha,
        parents: [latestCommitSha]
      })
      latestCommitSha = response.data.sha
      
      await octokit.git.updateRef({
        owner,
        repo,
        sha: latestCommitSha,
        ref: `heads/master`,
        force: true
      })
      
      console.log("Project saved");

    }
```

**Comment by gr2m** [comment]
Glad the code was useful!

We could create another Octokit plugin which extracts the logic of updating the files. `octokit-create-pull-request` could use that plugin internally.

What would we call that plugin? Maybe something like `octokit-multi-file-commit`?

**Comment by BarbourSmith** [comment]
Yeah, that sounds great. I think octokit-multi-file-commit explains what is going very clearly.

Would it be possible to support blobs to upload files larger than 1mb?

**Comment by gr2m** [comment]
It should be possible using https://developer-site4.githubapp.com/v3/git/blobs/#create-a-blob 🤔

**Comment by BarbourSmith** [comment]
Alright, so I am giving this another go after putting it off for a long time 😶 

I am thinking I need to 

```
octokit.git.createBlob({
  owner,
  repo,
  content
})
```

But I am a little unsure of how that would fit in with building the tree. 

Do you have any advice on where to start? 

Thanks! 😀

**Comment by gr2m** [comment]
I’m sorry I’m heads down with other work right now. This is quite complicated, I’d have to look it up myself. Could you try to figure it out? https://github.community/ can be a good resource, if not support@github.com might be able help, too

I should have more time next week. Keep me posted :)

**Comment by BarbourSmith** [comment]
Thanks for the advice, even just knowing that there isn't a one line solution makes a big difference. I will update you if I find a solution.

**Comment by d3netxer** [comment]
I don't get it, isn't this project already supposed to commit multiple files at once? In ReadMe it states: "Update or create two files with a single commit"

**Comment by gr2m** [comment]
I’ve changed the issue title to be more clear. In the issue description it says

> TLDR: Is there a way for me to modify this library to take n files and commit them to master instead of creating a pull request?

**Comment by BarbourSmith** [comment]
Thanks! Sorry for the unclear title 😶
